### PR TITLE
DX: new config filename

### PR DESCRIPTION
**Pull request by keradus**
old files are still allowed. will be disallowed with upgrade msg in #5609

### CODE REVIEW

**Comment by mvorisek** [comment]
I propose `.php_cs.php`

**Comment by keradus** [comment]
`dist` is common pattern for other tools, let's not introducing something new
`php_cs` was constantly confused with CodeSniffer, let's get rid of that once and for all

**Comment by mvorisek** [comment]
But `.local.` can be much easily ignored in `.gitignore`, ˙.php_cs˙ (or new equivalent) has priority over `dist`, but this name is very often not ignored thus not helpful. Let's think big and set the new standard 🚀 🍾

**Review by localheinz — APPROVED** [review]

**Comment by julienfalque** [comment]
Do we actually need to support having a "dist" file though? I never saw a valid use case and still fail to imagine one.

**Comment by mvorisek** [comment]
> 
> 
> Do we actually need to support having a "dist" file though? I never saw a valid use case and still fail to imagine one.

That is the case. We should be simple filename (without `dist`) and optionally support one prioritized name for local override.

**Comment by julienfalque** [comment]
> That is the case.

Please elaborate :)

**Comment by mvorisek** [comment]
> 
> 
> > That is the case.
> 
> Please elaborate :)

see above, `dist` was used to support local override, but naming is not optimal and thus requires this explanation :) Therefore I think https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5609/files#r609648455 is better and allow also more generic exclude, see https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5607#issuecomment-815204086

**Comment by julienfalque** [comment]
I know it allows local override, but why is that needed? The point of this tool is to enforce consistent code style between multiple code contributors. When would one need to diverge from that locally (i.e. apply a different code style)?

**Comment by mvorisek** [comment]
> 
> 
> I know it allows local override, but why is that needed? The point of this tool is to enforce consistent code style between multiple code contributors. When would one need to diverge from that locally (i.e. apply a different code style)?

I belive to support one DEFAULT local/prioritized name, ie. one name that does not have to be specified explicitly to override the config.

**Comment by keradus** [comment]
@julienfalque , i know you are not in favour of multiple config files approach ;)

what is `dist` for: https://stackoverflow.com/a/44764078
don't like dist, don't use it ;) https://github.com/symfony/recipes/issues/41#issuecomment-298343132

**Comment by KacerCZ** [comment]
@mvorisek Use of `dist` (short from 'distribute') configurations is common in PHP tools. I don't think it is good idea to change this pattern unless you want to change it in all other tools also.

@julienfalque My typical use case for local and `dist` config is introduction of new rules to project.
Repository contains only `dist` configuration and CI uses it to check code.
My local configuration is copy of `dist` with one or more rules enabled.
So when I modify code it allows me to fix more errors.
I don't have time to fix all errors at once (specially risky rules).

**Comment by mvorisek** [comment]
> want to change it in all other tools also

I would be happy to do so witch CS Fixer taking the lead :)

**Review by kubawerlos — APPROVED** [review]

File: src/Console/ConfigurationResolver.php [hunk 1]
```diff
@@ -227,6 +227,23 @@ public function getConfig()
...
+
+                if (isset($deprecatedConfigs[$configFileBasename])) {
+                    $message = "Configuration file `{$configFileBasename}` is deprecated, rename to `{$deprecatedConfigs[$configFileBasename]}`.";
+
+                    // TODO 3.0: always throw exception
+                    if (getenv('PHP_CS_FIXER_FUTURE_MODE')) {
+                        throw new InvalidConfigurationException("{$message} This check was performed as `PHP_CS_FIXER_FUTURE_MODE` env var is set.");
+                    }
+
+                    @trigger_error($message, E_USER_DEPRECATED);
```
Comment (keradus) [inline src/Console/ConfigurationResolver.php:244]: marker: use utils @keradus

**Pull request merged**